commands/project/create/project_create.go

```diff
@@ -199,6 +205,22 @@ func runCreateProject(cmd *cobra.Command, args []string, f *cmdutils.Factory) er
 	return err
 }
 
+func initGit() error {
+	if stat, err := os.Stat(".git"); err == nil && stat.IsDir() {
+		return nil
+	}
+	var doInit bool
+	err := prompt.Confirm(&doInit, fmt.Sprintf("Directory not git initialized. Run `git init`?"), true)
+	if err != nil || !doInit {
+		return err
+	}
+
+	gitInit := git.GitCommand("init")
```

Comment:
Don't you think we should consider `git init`-ializing with the default branch provided by the user via the `--defaultBranch` flag?

If so, the `initGit` function signature will change to `initGit(branch string)`.

```go
gitInit := git.GitCommand("init", "-b", branch)
```

On line 71, you would have to then call the `initGit` function with the default branch provided by the user
```go
err = initGit(cmd.Flags().GetString("defaultBranch"))
if err != nil {
	return err
}
```

Reply:
Yeah I totally agree. Didn't notice that a default branch could be passed. I'll make the change.

Reply:
Great!

Reply:
@profclems Instead of using the `-b` flag I have added a checkout command after git init if the default branch is not empty. This is to ensure backwards compatibility with older versions of git.

Reply:
Makes perfect sense! Thanks